### PR TITLE
add add-push-manifest option

### PR DESCRIPTION
**Pull request by FredKSchott**
- [X] CHANGELOG.md has been updated
- [x] Needs https://github.com/Polymer/polymer-project-config/pull/12
  - tests will not pass until that PR is merged and released
- [x] Needs https://github.com/Polymer/polymer-build/pull/172
  - tests will not pass until that PR is merged and released

### CODE REVIEW

File: src/build/build.ts [hunk 1]
```diff
@@ -88,6 +88,15 @@ export async function build(
     logger.info(`(${buildName}) Building...`);
   });
 
+  if (options.addPushManifest) {
+    buildStream = buildStream.pipe(polymerProject.addPushManifest());
+    if (!options.addServiceWorker) {
+      logger.warn('Push manifest generated without a service worker. ' +
+      'HTTP2/Push relies on service workers to prevent duplicate file ' +
```
Comment (rictic) [inline src/build/build.ts]: I did not know this.

Comment (justinfagnani) [inline src/build/build.ts]: I didn't either. I thought that the browser would cancel pushes if it already had the file in the cache.

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,9 @@
 
 <!--## Unreleased-->
 
+- `build`: Added `--app-push-manifest`/`addPushManifest` option for generating a [`push-manifest.json`](https://github.com/GoogleChrome/http2-push-manifest) file for your project.
```
Comment (tmlbl) [inline CHANGELOG.md]: Typo in flag name

Comment (FredKSchott) [inline CHANGELOG.md]: doh, thanks for catching that 👍

File: src/build/build.ts (hunk 1 above)
Comment (FredKSchott) [inline src/build/build.ts]: I looked into this more and you guys are right, there are a few different ways for the browser/server to negotiate to stop duplicate pushes. Removed in 2708c9b.

**Comment by justinfagnani** [comment]
@FredKSchott merge/rebase?

**Comment by FredKSchott** [comment]
@justinfagnani done, and dependencies updated. TAL?

**Pull request merged**